docker-compose-core.yaml

```
@@ -116,9 +124,48 @@ services:
       - zeebe
       - operate
 
+  opensearch:
+    image: opensearchproject/opensearch:${OPENSEARCH_VERSION}
+    container_name: opensearch
+    profiles: ["opensearch"]
+    depends_on:
+      - init
+    environment:
+      cluster.name: opensearch-cluster
+      network.bind_host: 0.0.0.0
+      node.name: opensearch
+      plugins.security.disabled: true
+      plugins.security.audit.config.index: myauditlogindex
+      discovery.type: single-node
+      discovery.seed_hosts: opensearch
+      bootstrap.memory_lock: true
+      OPENSEARCH_JAVA_OPTS: "-Xms1G -Xmx1G"
+      OPENSEARCH_INITIAL_ADMIN_PASSWORD: ${OPENSEARCH_INITIAL_ADMIN_PASSWORD}
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    ports:
+      - "9200:9200"
+      - "9601:9600"
+    restart: always
+    healthcheck:
+      test: [ "CMD-SHELL", "curl -k -f http://localhost:9200/_cluster/health -u admin:${OPENSEARCH_INITIAL_ADMIN_PASSWORD} | grep -q -E 'yellow|green'" ]
```

Suggesting to leave a note why yellow is okay-ish
```suggestion
      # Single node OpenSearch clusters are considered 'yellow', see https://github.com/opensearch-project/opensearch-build/issues/4285
      test: [ "CMD-SHELL", "curl -k -f http://localhost:9200/_cluster/health -u admin:${OPENSEARCH_INITIAL_ADMIN_PASSWORD} | grep -q -E 'yellow|green'" ]
```

---

Agree with that suggestion